src/wp-includes/blocks/page-list.php

```diff
@@ -0,0 +1,227 @@
+		// Add has-background class.
+		$colors['css_classes'][] = 'has-background';
+	}
+
+	if ( $has_named_background_color ) {
+		// Add the background-color class.
+		$colors['css_classes'][] = sprintf( 'has-%s-background-color', $context['backgroundColor'] );
+	} elseif ( $has_custom_background_color ) {
+		// Add the custom background-color inline style.
+		$colors['inline_styles'] .= sprintf( 'background-color: %s;', $context['style']['color']['background'] );
+	}
+
+	return $colors;
+}
+
+/**
+ * Build an array with CSS classes and inline styles defining the font sizes
+ * which will be applied to the pages markup in the front-end when it is a descendant of navigation.
+ *
+ * @param  array $context Navigation block context.
+ * @return array Font size CSS classes and inline styles.
+ */
+function block_core_page_list_build_css_font_sizes( $context ) {
+	// CSS classes.
+	$font_sizes = array(
+		'css_classes'   => array(),
+		'inline_styles' => '',
+	);
+
+	$has_named_font_size  = array_key_exists( 'fontSize', $context );
+	$has_custom_font_size = isset( $context['style']['typography']['fontSize'] );
+
+	if ( $has_named_font_size ) {
+		// Add the font size class.
+		$font_sizes['css_classes'][] = sprintf( 'has-%s-font-size', $context['fontSize'] );
+	} elseif ( $has_custom_font_size ) {
+		// Add the custom font size inline style.
+		$font_sizes['inline_styles'] = sprintf( 'font-size: %spx;', $context['style']['typography']['fontSize'] );
+	}
+
+	return $font_sizes;
+}
+
+/**
+ * Outputs Page list markup from an array of pages with nested children.
+ *
+ * @param array $nested_pages The array of nested pages.
+ *
+ * @return string List markup.
+ */
+function render_nested_page_list( $nested_pages ) {
+	if ( empty( $nested_pages ) ) {
+		return;
+	}
+	$markup = '';
+	foreach ( (array) $nested_pages as $page ) {
+		$css_class = 'wp-block-pages-list__item';
+		if ( isset( $page['children'] ) ) {
+			$css_class .= ' has-child';
+		}
+		$markup .= '<li class="' . $css_class . '">';
+		$markup .= '<a class="wp-block-pages-list__item__link" href="' . esc_url( $page['link'] ) . '">' . wp_kses(
+			$page['title'],
+			wp_kses_allowed_html( 'post' )
+		) . '</a>';
+		if ( isset( $page['children'] ) ) {
+			$markup .= '<span class="wp-block-page-list__submenu-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="12" viewBox="0 0 12 12" fill="none" role="img" aria-hidden="true" focusable="false"><path d="M1.50002 4L6.00002 8L10.5 4" stroke-width="1.5"></path></svg></span>';
+			$markup .= '<ul class="submenu-container">' . render_nested_page_list( $page['children'] ) . '</ul>';
+		}
+		$markup .= '</li>';
+	}
+	return $markup;
+}
+
+/**
+ * Outputs nested array of pages
+ *
+ * @param array $current_level The level being iterated through.
+ * @param array $children The children grouped by parent post ID.
+ *
+ * @return array The nested array of pages.
+ */
+function nest_pages( $current_level, $children ) {
```

Comment:
This is not ideally named but we should follow-up on this in Gutenberg. This file is directly ported (using webpack copy)